### PR TITLE
Muq 34

### PR DESCRIPTION
**Pull request by mnzit**

### CODE REVIEW

**Comment by github-gada** [comment]
Author is not in whitelist. Can one of the admins verify this patch?

File: src/main/java/life/genny/qwanda/converter/MinIOConverter.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public String convertToDatabaseColumn(String attribute) {
+        return attribute;
+    }
+
+    @Override
+    public String convertToEntityAttribute(String dbData) {
+        if (dbData != null && dbData.startsWith(QwandaConstant.MINIO_LAZY_PREFIX)) {
+            log.info("####### Fetching from MinIO");
```
Comment (varunshastry) [inline src/main/java/life/genny/qwanda/converter/MinIOConverter.java]: ####### to be removed

File: src/main/java/life/genny/qwanda/util/minio/Minio.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                            MinIOEnvironmentVariable.MINIO_PRIVATE_KEY);
+        } catch (InvalidEndpointException e) {
+            e.printStackTrace();
+        } catch (InvalidPortException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String saveOnStore(FileUpload file) {
+        if (uploadFile(REALM.concat("/") + "public", file.uploadedFileName(), file.fileName())) {
```
Comment (varunshastry) [inline src/main/java/life/genny/qwanda/util/minio/Minio.java]: Extract the function call to local variable for better clarity.

File: src/main/java/life/genny/qwanda/util/minio/Minio.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            return file.fileName();
+        } else {
+            return null;
+        }
+    }
+
+    public static UUID saveOnStore(FileUpload file, UUID userUUID) {
+        UUID randomUUID = UUID.randomUUID();
+        if (uploadFile(userUUID.toString(), file.uploadedFileName(), randomUUID.toString())) {
```
Comment (varunshastry) [inline src/main/java/life/genny/qwanda/util/minio/Minio.java]: Extract the function call to local variable for better clarity.

File: src/main/java/life/genny/qwanda/util/minio/Minio.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            InputStream object = minioClient.getObject(MinIOEnvironmentVariable.BUCKET_NAME,
+                    userUUID.toString() + "/media/" + fileUUID.toString());
+            byte[] byteArray = IOUtils.toByteArray(object);
+            return byteArray;
+        } catch (InvalidKeyException | InvalidBucketNameException
+                 | NoSuchAlgorithmException | InsufficientDataException
+                 | NoResponseException | ErrorResponseException
+                 | InternalException | InvalidArgumentException | IOException
+                 | XmlPullParserException e) {
+            e.printStackTrace();
```
Comment (varunshastry) [inline src/main/java/life/genny/qwanda/util/minio/Minio.java]: Will printstacktrace log the exception to log file as "INFO"? Do we want to add specific message?

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: src/main/java/life/genny/qwanda/util/minio/MinIOEnvironmentVariable.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  public final static String MINIO_SERVER_URL;
+  public final static String MINIO_ACCESS_KEY;
+  public final static String MINIO_PRIVATE_KEY;
+  public final static String BUCKET_NAME;
+
+  static {
+    /* Values from Environment */
+    Optional<String> minioServerURL =
+        Optional.ofNullable(System.getenv("MINIO_SERVER_URL"));
```
Comment (varunshastry) [inline src/main/java/life/genny/qwanda/util/minio/MinIOEnvironmentVariable.java]: Extract to "Constants"

**Review by varunshastry — CHANGES_REQUESTED** [review]

**Pull request merged**